lib/parser/base.rb

```diff
@@ -210,8 +210,9 @@ def parse_with_comments(source_buffer)
 
     ##
     # Parses a source buffer and returns the AST, the source code comments,
-    # and the tokens emitted by the lexer. If `recover` is true and a fatal
-    # {SyntaxError} is encountered, `nil` is returned instead of the AST, and
+    # and the tokens emitted by the lexer. In case of a fatal error, a {SyntaxError}
```

Comment:
ha, it took me a while to understand that `recover` here is responsible for "swallowing the error", but not "error recovery". Not sure if it's worth renaming it, current name sounds good as a part of the public API (you "recover" from VM error, not from the parsing error). Thanks for changing this doc too 👍